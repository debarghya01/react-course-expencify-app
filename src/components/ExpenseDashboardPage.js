import React from 'react';
import ReactDOM from 'react-dom';
import ExpensList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';


const ExpenseDashboardPage = () => {
    return(
        <div>
            <ExpenseListFilters />
            <ExpensList />
        </div>
    )
}

export default ExpenseDashboardPage;