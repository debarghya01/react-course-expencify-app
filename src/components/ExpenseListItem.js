import React from 'react';
import { Link } from 'react-router-dom';


const ExpenseListItem = (props) => {
    
    return(
        <div>
            <Link to={`/edit/${props.id}`} ><p>  {`Expense ${props.description}`}</p> </Link> <p>{`${props.amount} ${props.createdAt}`} </p> 

        </div>
    )
}

export default ExpenseListItem;
