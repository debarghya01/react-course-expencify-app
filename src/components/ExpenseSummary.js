import React from 'react';
import {connect} from 'react-redux';
import numeral from 'numeral';
import getVisibleExpenses from './../selectors/expenses';
import getExpenseTotal from './../selectors/expenses-total';

export const ExpenseSummary = (props) => {
    return(
        <div>
            <h2>
                {`Viewing ${props.count} expenses totaling ${props.amount}`} 
            </h2>
        </div>
    )
} 

const mapStateToProps = (state) => {
    const expenses = getVisibleExpenses(state.expenses, state.filters)
    let total = getExpenseTotal(expenses)
    total = total/100;
    const formatedTotal = numeral(total).format('$0,0.00');
    return(
        {
            count: expenses.length,
            amount: formatedTotal
        }
    )
}

const ConnectedExpenseSummary = connect(mapStateToProps)(ExpenseSummary);
export default ConnectedExpenseSummary;