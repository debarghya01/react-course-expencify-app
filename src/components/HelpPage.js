import React from 'react';
import RecatDOM from 'react-dom';

const HelpPage = () => {
    return(
        <div>Help page</div>
    )
}

export default HelpPage;
