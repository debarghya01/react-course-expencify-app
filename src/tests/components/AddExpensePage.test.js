import React from 'react';
import {AddExpensePage} from '../../components/AddExpensePage';
import {shallow} from 'enzyme';
import ExpenseForm from '../../components/ExpenseForm';
import Expenses from '../fixtures/expenses';

let addExpense, history, wrapper;
beforeEach(() => {
    addExpense = jest.fn()
    history = {push: jest.fn()}
    wrapper = shallow(<AddExpensePage addExpense={addExpense} history={history} />)
}) 
test('Testing add expense page render', () => {
        expect(wrapper).toMatchSnapshot()
})

test('Testing add expense page render with callback test', () => {
    wrapper.find(ExpenseForm).prop('onSubmit')(Expenses[0])
    expect(history.push).toHaveBeenLastCalledWith('/')
    expect(addExpense).toHaveBeenLastCalledWith(Expenses[0])
})
