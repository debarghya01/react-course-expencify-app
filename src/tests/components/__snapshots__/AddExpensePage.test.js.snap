// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing add expense page render 1`] = `
<div>
  <h1>
    Add expence
  </h1>
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
